web-frame/src/app-shared/core/initializers/app-features-initializer.service.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright (c) 2020 THREEANGLE SOFTWARE SOLUTIONS SRL
+ * Available under MIT license webFrame/LICENSE
+ */
+import { Injectable } from '@angular/core';
+import { Route, Router, Routes } from '@angular/router';
+import { Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { AVAILABLE_FEATURE_MODULES, ENVIRONMENT } from '~app-shared/config';
+
+@Injectable()
+export class AppFeaturesInitializerService {
+  constructor(
+    private router: Router,
+  ) {
+  }
+
+  public initialize(): Observable<void> {
+    return this.getEnabledFeatures().pipe(
+      map((enabledFeatures: string[]): void => {
+        const routes: Routes = this.getRoutesForEnabledFeatures(enabledFeatures);
+        this.registerAvailableRoutes(routes);
```

Comment:
By the time this method (`AppFeaturesInitializerService.initialize()`) is called, the state service already contains the user information and things like the user role can be used to further filter down the list of enabled features.